src/libtls/tls_peer.c

```diff
@@ -1405,33 +1399,32 @@ static status_t send_certificate(private_tls_peer_t *this,
 
 	version_min = this->tls->get_version_min(this->tls);
 	version_max = this->tls->get_version_max(this->tls);
-	if (version_max > TLS_1_1)
+	if (!this->hashsig.len)
+	{
+		convert_cert_types(this);
+	}
+	enumerator = tls_create_private_key_enumerator(version_min, version_max,
+												   this->hashsig, this->peer);
+	if (!enumerator || !enumerator->enumerate(enumerator, &key, &auth))
 	{
-		enumerator = tls_create_private_key_enumerator(version_min, version_max,
-													   this->hashsig, this->peer);
 		if (!enumerator)
 		{
 			DBG1(DBG_TLS, "no common signature algorithms found");
-			return FALSE;
 		}
-		if (enumerator->enumerate(enumerator, &key, &auth))
+		else
 		{
-			this->private = key->get_ref(key);
-			this->peer_auth->merge(this->peer_auth, auth, FALSE);
+			DBG1(DBG_TLS, "no usable TLS client certificate found for '%Y'",
+				 this->peer);
 		}
-		enumerator->destroy(enumerator);
+		this->peer->destroy(this->peer);
+		this->peer = NULL;
 	}
 	else
 	{
-		this->private = find_private_key(this);
-	}
-	if (!this->private)
-	{
-		DBG1(DBG_TLS, "no usable TLS client certificate found for '%Y'",
-			 this->peer);
-		this->peer->destroy(this->peer);
-		this->peer = NULL;
+		this->private = key->get_ref(key);
+		this->peer_auth->merge(this->peer_auth, auth, FALSE);
 	}
+	enumerator->destroy(enumerator);
```

Comment:
Here an additional check is needed:

```
if (enumerator)
{
    enumerator->destroy(enumerator);
}
```

Reply:
That is, just use `DESTROY_IF(enumerator);`.